pallets/subtensor/src/swap.rs

```diff
@@ -0,0 +1,422 @@
+    ///
+    /// # Returns
+    ///
+    /// * `Result<Vec<u16>, Error<T>>` - A vector of network IDs where the hotkey is a member.
+    pub fn get_netuid_is_member(
+        old_hotkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> Result<Vec<u16>, Error<T>> {
+        let netuid_is_member: Vec<u16> =
+            <IsNetworkMember<T> as IterableStorageDoubleMap<_, _, _>>::iter_prefix(old_hotkey)
+                .map(|(netuid, _)| netuid)
+                .collect();
+        weight.saturating_accrue(T::DbWeight::get().reads(netuid_is_member.len() as u64));
+        Ok(netuid_is_member)
+    }
+
+    /// Swaps the owner of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `coldkey` - The coldkey owning the hotkey.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_owner(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        coldkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        Owner::<T>::remove(old_hotkey);
+        Owner::<T>::insert(new_hotkey, coldkey.clone());
+        weight.saturating_accrue(T::DbWeight::get().writes(2));
+        Ok(())
+    }
+
+    /// Swaps the total stake of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_total_hotkey_stake(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        if let Ok(total_hotkey_stake) = TotalHotkeyStake::<T>::try_get(old_hotkey) {
+            TotalHotkeyStake::<T>::remove(old_hotkey);
+            TotalHotkeyStake::<T>::insert(new_hotkey, total_hotkey_stake);
+            weight.saturating_accrue(T::DbWeight::get().writes(2));
+        }
+        Ok(())
+    }
+
+    /// Swaps the delegates of the hotkey.
+    ///
+    /// # Arguments
+    ///
+    /// * `old_hotkey` - The old hotkey.
+    /// * `new_hotkey` - The new hotkey.
+    /// * `weight` - The weight of the transaction.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), Error<T>>` - The result of the operation.
+    pub fn swap_delegates(
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+        weight: &mut Weight,
+    ) -> DispatchResult {
+        if let Ok(delegate_take) = Delegates::<T>::try_get(old_hotkey) {
+            Delegates::<T>::remove(old_hotkey);
+            Delegates::<T>::insert(new_hotkey, delegate_take);
+            weight.saturating_accrue(T::DbWeight::get().writes(2));
+        }
+        Ok(())
```

Comment:
Missing weight calculation from `try_get`, and doesn't really need to return a `DispatchResult`.

Reply:
done